src/test/java/no/unit/nva/cristin/projects/FetchCristinProjectsTest.java

```diff
@@ -450,6 +472,62 @@ void handlerReturnsProjectsWrapperWithCorrectNextResultsAndPreviousResultsWhenLi
         assertEquals(expectedPrevious, actualPrevious);
     }
 
+    @Test
+    void handlerReturnsMatchingProjectsFromGrantIdSearchWhenSuppliedWithOnlyNumber() throws Exception {
+        cristinApiClientStub = spy(cristinApiClientStub);
+
+        doReturn(new HttpResponseStub(CRISTIN_QUERY_PROJECTS_RESPONSE_JSON_FILE))
+            .when(cristinApiClientStub).queryProjects(any(), eq(QUERY_USING_GRANT_ID));
+
+        doThrow(RuntimeException.class)
+            .when(cristinApiClientStub).queryProjects(any(), eq(QUERY_USING_TITLE));
+
+        InputStream input = requestWithQueryParameters(Map.of(QUERY, GRANT_ID_EXAMPLE));
+        handler.handleRequest(input, output, context);
+        GatewayResponse<ProjectsWrapper> gatewayResponse = GatewayResponse.fromOutputStream(output);
+
+        assertEquals(HttpURLConnection.HTTP_OK, gatewayResponse.getStatusCode());
+        assertEquals(MediaType.APPLICATION_JSON, gatewayResponse.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    void handlerReturnsProjectsFromTitleSearchWhenSuppliedWithQueryStringIncludingNumber() throws Exception {
+        cristinApiClientStub = spy(cristinApiClientStub);
+
+        doThrow(RuntimeException.class)
```

Comment:
I can change the input value to anything and the tests succeed any time. I can delete the first mocking statement and the tests are succeeding again.

I think the sequence of requests is not tested in the tests and I think this is what we are interested in. The exception that is thrown by the mock is not observed in any way. I would suggest to verify that the exception is thrown and that afterwards the response is received. Otherwise I can change the inputStream of the test and the test will succeed every time.

Reply:
The same holds for all tests

Reply:
Must have made som errors writing those tests, i will fix

Reply:
See latest commit b8a752d

Reply:
It throws an exception if a query using grant id is called and if that happens the test should fail. So if we dont get an exception, it means it only queries using title which it should. It got updated in latest commit.